src/main/java/uk/gov/admin/DataFileReader.java

```diff
@@ -1,34 +1,39 @@
 package uk.gov.admin;
 
-import com.fasterxml.jackson.databind.MappingIterator;
+import com.fasterxml.jackson.core.type.TypeReference;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.dataformat.csv.CsvMapper;
 import com.fasterxml.jackson.dataformat.csv.CsvSchema;
 import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
 import com.google.common.collect.Iterators;
 
-import java.io.BufferedReader;
-import java.io.File;
-import java.io.IOException;
-import java.io.InputStreamReader;
+import java.io.*;
+import java.net.MalformedURLException;
 import java.net.URI;
 import java.net.URISyntaxException;
+import java.net.URL;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
-import java.util.Iterator;
-import java.util.Map;
+import java.util.*;
+import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
+import static java.lang.String.valueOf;
+import static java.util.stream.Collectors.toMap;
+
 public class DataFileReader {
     private final URI dataSourceURI;
     private final String type;
     private final ObjectMapper yamlObjectMapper = new ObjectMapper(new YAMLFactory());
+    private final ObjectMapper jsonMapper = new ObjectMapper();
+    private final Optional<Map<String, String>> cardinalities;
 
-    public DataFileReader(String dataSource, String type) throws URISyntaxException {
+    public DataFileReader(String dataSource, String type, Optional<String> fieldsSource) throws URISyntaxException {
```

Comment:
maybe constructor chaining?
```
public DataFileReader(String dataSource, String type) throws URISyntaxException {
        this(dataSource, type, Optional.empty());
    }
```
and then we don't have to add Optional.empty() in tests etc